File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        if sum(nums) < target:
            return 0
        if nums[0] >= target:
            return 1
        nums_sum = nums[0]
        ans = float("inf")
        left = 0
        for i in range(1, len(nums)):
            nums_sum += nums[i]       
            while nums_sum >= target:
                ans = min(ans, i - left + 1)
                nums_sum -= nums[left]
                left +=1
        return ans

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let sumWindow = 0;
    let minSub = Infinity;
    let left = 0;
    for (let i = 0; i < nums.length; i++) {
        sumWindow += nums[i];
        while (sumWindow >= target) {
            minSub = Math.min(minSub, i-left+1);
            sumWindow -= nums[left++];
        }
    }
    return minSub === Infinity ? 0: minSub;

};